.github/workflows/publish-code.yml

```diff
@@ -0,0 +1,74 @@
+name: Publish release
+
+on:
+  release:
+    types:
+      - published
+
+env:
+  NODE_VERSION: '10.x'   # Node 10 LTS
+
+jobs:
+  publish:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - name: Check github.ref starts with 'refs/tags/'
+        if: ${{ !startsWith(github.ref, 'refs/tags/') }}
+        run: |
+          echo Error! github.ref does not start with 'refs/tags'
+          echo github.ref: ${{ github.ref }}
+          exit 1
+      - name: Set version number environment variable
+        env:
+          github_ref: ${{ github.ref }}
+        run: |
+          ref="${github_ref:10}"
```

Comment:
What is `ref`? Action variable? Why do you need to first create an environment variable, put the tag value into it, and then transfer it to the `ref`? Also what is `:10` ?

Reply:
the C# equivalent would be:
```
var ref = github_ref.Substring(10);
```
basically, strip off the first 10 characters, and store in "ref"

Reply:
You'll also notice the following line which is required to store the temporary environment variable as permanent for the rest of the workflow:
```
echo "::set-env name=ref::$ref"
```
See https://docs.github.com/en/free-pro-team@latest/actions/reference/workflow-commands-for-github-actions

Reply:
It's possible that something like this could work:
```
echo "::set-env name=ref::${github_ref:10}"
```
I have not tried that.